database-commons/src/main/java/co/cask/db/batch/source/DBSource.java

```diff
@@ -243,79 +239,69 @@ public void initialize(BatchRuntimeContext context) throws Exception {
   @Override
   public void transform(KeyValue<LongWritable, DBRecord> input, Emitter<StructuredRecord> emitter) throws Exception {
     emitter.emit(StructuredRecordUtils.convertCase(
-      input.getValue().getRecord(), FieldCase.toFieldCase(sourceConfig.columnNameCase)));
+      input.getValue().getRecord(), FieldCase.NONE));
   }
 
   @Override
   public void destroy() {
-    try {
-      DBUtils.cleanup(driverClass);
-    } finally {
-      dbManager.destroy();
-    }
+    DBUtils.cleanup(driverClass);
   }
 
   private String getJDBCPluginId() {
-    return String.format("%s.%s.%s", "source", sourceConfig.jdbcPluginType, sourceConfig.jdbcPluginName);
+    return String.format("%s.%s.%s", "source", ConnectionConfig.JDBC_PLUGIN_TYPE, getJdbcDriverName());
   }
 
   /**
    * {@link PluginConfig} for {@link DBSource}
    */
-  public static class DBSourceConfig extends DBConfig {
+  public abstract static class DBSourceConfig extends DBConfig {
     public static final String IMPORT_QUERY = "importQuery";
     public static final String BOUNDING_QUERY = "boundingQuery";
     public static final String SPLIT_BY = "splitBy";
     public static final String NUM_SPLITS = "numSplits";
     public static final String SCHEMA = "schema";
     public static final String TRANSACTION_ISOLATION_LEVEL = "transactionIsolationLevel";
-
+    @Nullable
+    @Name(TRANSACTION_ISOLATION_LEVEL)
+    @Description("The transaction isolation level for queries run by this sink. " +
+      "Defaults to TRANSACTION_SERIALIZABLE. See java.sql.Connection#setTransactionIsolation for more details. " +
```

Comment:
it looks like this was added specifically for phoenix. This should not show up for the non-generic db plugins.

Does it make sense to have it for mysql?

Reply:
It's not required by MySQL though a lot of code relies on transactionIsolationLevel ifrom DBSource and DBSink configs. It's not possible to move it from DBSinkConfig since  OutputFormatProvider expects it to be there. Although, we can completely remove the dedicated TRANSACTION_ISOLATION property and specify it as part of advanced settings via key-value fields.

Reply:
It sounds like it doesn't belong in MySQL, which means it shouldn't be in the base config. The DB specific configs should not extend the config for the generic DB plugin. They should all extend a common parent that only has the properties that are exactly the same (descriptions included) for all of them.